internal/pkg/server/server.go

```diff
@@ -82,6 +120,201 @@ func (s *Server) postTar(c echo.Context) error {
 	return s.streamImages(c, authn, normalizeImages(images))
 }
 
+func (s *Server) postHelmRepo(c echo.Context) error {
+	name := c.FormValue("name")
+	url := c.FormValue("url")
+	if name == "" {
+		return fmt.Errorf("repository name not set")
+	}
+	if url == "" {
+		return fmt.Errorf("repository URL not set")
+	}
+
+	// The following code is adapted from https://github.com/helm/helm/blob/master/cmd/helm/repo_add.go
+	fileLock := flock.New(strings.Replace(repoFile, filepath.Ext(repoFile), ".lock", 1))
+	lockCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	locked, err := fileLock.TryLockContext(lockCtx, time.Second)
+	if err == nil && locked {
+		defer fileLock.Unlock()
+	}
+	if err != nil {
+		return err
+	}
+
+	b, err := ioutil.ReadFile(repoFile)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	var f repo.File
+	if err := yaml.Unmarshal(b, &f); err != nil {
+		return err
+	}
+
+	cfg := repo.Entry{
+		Name:                  name,
+		URL:                   url,
+		Username:              c.FormValue("username"),
+		Password:              c.FormValue("password"),
+		InsecureSkipTLSverify: false,
+	}
+	repo, err := repo.NewChartRepository(&cfg, getters)
+	if err != nil {
+		return err
+	}
+	if repoCache != "" {
+		repo.CachePath = repoCache
+	}
+
+	if _, err := repo.DownloadIndexFile(); err != nil {
+		return errors.Wrapf(err, "looks like %q is not a valid chart repository or cannot be reached", url)
+	}
+
+	f.Update(&cfg)
+	if err := f.WriteFile(repoFile, 0644); err != nil {
+		return err
+	}
+
+	c.Logger().Info(fmt.Sprintf("Repository addedd successfully: %s %s", name, url))
+	return nil
+}
+
+func (s *Server) postHelmChart(c echo.Context) error {
+	chrtRef := c.FormValue("chart")
+	chrtVer := c.FormValue("version")
+	file, err := c.FormFile("values.yaml")
+	if err != nil {
+		return err
+	}
+	src, err := file.Open()
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+	buf, err := ioutil.ReadAll(src)
+	if err != nil {
+		return err
+	}
+	values := map[string]interface{}{}
+	if err := yaml.Unmarshal(buf, &values); err != nil {
+		return err
+	}
+	var authn auth.Authenticator
+	_, ok := c.QueryParams()["auth"] // Currently support only docker config.json
```

Comment:
What's the reasoning behind the extra auth parameter? `POST /tar` always takes the `config.json` file from the request or the one stored on disk. I think it's a bit counter intuitive to require the parameter for one endpoint but not the other.

Reply:
Totally, thanks